### PR TITLE
squeezelite: add darwin support

### PR DESCRIPTION
**Pull request by adamcstephens**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by adamcstephens** [comment]
esult of `nixpkgs-review pr 232452` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>squeezelite</li>
    <li>squeezelite-pulse</li>
  </ul>
</details>

-----

Result of `nixpkgs-review pr 232452` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>squeezelite</li>
    <li>squeezelite-pulse</li>
  </ul>
</details>

-----

Result of `nixpkgs-review pr 232452` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>squeezelite</li>
  </ul>
</details>

-----

Result of `nixpkgs-review pr 232452` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>squeezelite</li>
  </ul>

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -34148,9 +34148,15 @@ with pkgs;
 
   psst = callPackage ../applications/audio/psst { };
 
-  squeezelite = callPackage ../applications/audio/squeezelite { audioBackend = "alsa"; };
+  squeezelite = darwin.apple_sdk_11_0.callPackage ../applications/audio/squeezelite {
+    inherit (darwin.apple_sdk.frameworks) CoreVideo VideoDecodeAcceleration CoreAudio AudioToolbox AudioUnit Carbon;
```
Comment (wegank) [inline pkgs/top-level/all-packages.nix:34152]:
```suggestion
    inherit (darwin.apple_sdk_11_0.frameworks) CoreVideo VideoDecodeAcceleration CoreAudio AudioToolbox AudioUnit Carbon;
```

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -34148,9 +34148,15 @@ with pkgs;
...
+  };
+
+  squeezelite-pulse = darwin.apple_sdk_11_0.callPackage ../applications/audio/squeezelite {
+    inherit (darwin.apple_sdk.frameworks) CoreVideo VideoDecodeAcceleration CoreAudio AudioToolbox AudioUnit Carbon;
```
Comment (wegank) [inline pkgs/top-level/all-packages.nix:34156]:
```suggestion
    inherit (darwin.apple_sdk_11_0.frameworks) CoreVideo VideoDecodeAcceleration CoreAudio AudioToolbox AudioUnit Carbon;
```

File: pkgs/applications/audio/squeezelite/default.nix [hunk 3]
```diff
@@ -65,10 +77,15 @@ stdenv.mkDerivation {
     ++ optional (!faad2Support) "-DNO_FAAD"
     ++ optional ffmpegSupport "-DFFMPEG"
     ++ optional opusSupport "-DOPUS"
+    ++ optional portaudioSupport "-DPORTAUDIO"
     ++ optional pulseSupport "-DPULSEAUDIO"
     ++ optional resampleSupport "-DRESAMPLE"
     ++ optional sslSupport "-DUSE_SSL";
 
+  LDADD = if stdenv.isLinux
```
Comment (wegank) [inline pkgs/applications/audio/squeezelite/default.nix:85]: Should probably be set to `null` if it is not needed on Linux.

File: pkgs/applications/audio/squeezelite/default.nix [hunk 4]
```diff
@@ -44,7 +53,10 @@ stdenv.mkDerivation {
   };
 
   buildInputs = [ flac libmad libvorbis mpg123 ]
-    ++ lib.singleton (if pulseSupport then libpulseaudio else alsa-lib)
+    ++ optional pulseSupport libpulseaudio
+    ++ optional alsaSupport alsa-lib
+    ++ optional portaudioSupport portaudio
+    ++ optionals stdenv.isDarwin [CoreVideo VideoDecodeAcceleration CoreAudio AudioToolbox AudioUnit Carbon]
```
Comment (wegank) [inline pkgs/applications/audio/squeezelite/default.nix:59]:
```suggestion
    ++ optionals stdenv.isDarwin [ CoreAudio VideoDecodeAcceleration AudioToolbox AudioUnit Carbon ]
```

Comment (adamcstephens) [reply pkgs/applications/audio/squeezelite/default.nix]: Did you mean to change the frameworks here, or just add space padding inside the list?

Comment (adamcstephens) [reply pkgs/applications/audio/squeezelite/default.nix]: Framework list lifted from upstream here: https://github.com/ralph-irving/squeezelite/blob/575b59398dd19945d2038474062240fc527407fb/Makefile.m1#L6

Comment (wegank) [reply pkgs/applications/audio/squeezelite/default.nix]: Nope, just to remove the duplicated `CoreAudio` here.

Comment (wegank) [reply pkgs/applications/audio/squeezelite/default.nix]: Oh, my bad.

**Comment by wegank** [comment]
@ofborg build squeezelite squeezelite.passthru.tests

**Review by wegank — APPROVED** [review]

**Comment by adamcstephens** [comment]
Thanks for your assistance!

**Pull request merged**